get_e2e_test.go

```diff
@@ -603,6 +609,10 @@ func TestGet(t *testing.T) {
 						// Regression test against https://github.com/bwplotka/bingo/issues/65.
 						name: "get tool with capital letters in name (pre modules)",
 						do: func(t *testing.T) {
+							if !goVersion.LessThan(version.Go116) {
```

Comment:
Do we know why this happens? 🤔 

````
    get_e2e_test.go:612: get_e2e_utils_test.go:229:
        
         unexpected error: error while running command "/tmp/bingo-tmpgoenv193949100/bingo get github.com/githubnemo/CompileDaemon@v1.2.1"; out: Error: get command failed: get: compiledaemon.mod: getting github.com/githubnemo/CompileDaemon@v1.2.1: install: build versioned: ../../gopath/pkg/mod/github.com/githubnemo/!compile!daemon@v1.2.1/daemon.go:79:2: no required module provides package github.com/fatih/color; to add it:
        	go get github.com/fatih/color
        ../../gopath/pkg/mod/github.com/githubnemo/!compile!daemon@v1.2.1/daemon.go:80:2: no required module provides package github.com/fsnotify/fsnotify; to add it:
        	go get github.com/fsnotify/fsnotify
        : exit 1
        ; err: exit status 1
        
            --- FAIL: TestGet/empty_project_with_advanced_cases/isGoProject=false/get_tool_with_capital_letters_in_name_(pre_modules) (5.90s) 
    ```

Reply:
Actually it's more complex:

```
  
         unexpected error: error while running command "/tmp/bingo-tmpgoenv281836461/bingo get -v github.com/githubnemo/CompileDaemon@v1.2.1"; out: getting target github.com/githubnemo/CompileDaemon@v1.2.1 (module  )
        exec 'go mod init -modfile=/tmp/bingo-tmpgoenv281836461/testproject/.bingo/compiledaemon-e.tmp.mod _'
        exec 'go get -modfile=/tmp/bingo-tmpgoenv281836461/testproject/.bingo/compiledaemon-e.tmp.mod -d github.com/githubnemo/CompileDaemon@v1.2.1'
        exec 'go env GOPATH'
        exec 'go mod init -modfile=/tmp/bingo-tmpgoenv281836461/testproject/.bingo/compiledaemon.tmp.mod _'
        exec 'go mod download -x -modfile=/tmp/bingo-tmpgoenv281836461/testproject/.bingo/compiledaemon.tmp.mod'
        exec 'go list -modfile=/tmp/bingo-tmpgoenv281836461/testproject/.bingo/compiledaemon.tmp.mod -f={{.Name}} github.com/githubnemo/CompileDaemon'
        Error: install: build versioned: ../../gopath/pkg/mod/github.com/githubnemo/!compile!daemon@v1.2.1/daemon.go:79:2: no required module provides package github.com/fatih/color; to add it:
        	go get github.com/fatih/color
        ../../gopath/pkg/mod/github.com/githubnemo/!compile!daemon@v1.2.1/daemon.go:80:2: no required module provides package github.com/fsnotify/fsnotify; to add it:
        	go get github.com/fsnotify/fsnotify
        : error while running command 'go build -modfile=/tmp/bingo-tmpgoenv281836461/testproject/.bingo/compiledaemon.tmp.mod -o=/tmp/bingo-tmpgoenv281836461/bin/compiledaemon-v1.2.1 github.com/githubnemo/CompileDaemon'; err: exit status 1
        compiledaemon.mod: getting github.com/githubnemo/CompileDaemon@v1.2.1
        main.get
```

Reply:
Found root cause: 
go mod download does not resolve anything, so in go.mod we have only direct dep. Only `go get` adds `go.sum` files for all deps too.

Reply:
I also found that the current code does not work well in some cases, such as commits specified directly. This one is failing as well:
```
$ bingo -v get github.com/envoyproxy/protoc-gen-validate@d6164de4910977d3c3c8dbd9299b5064ea9e7c2b

getting target github.com/envoyproxy/protoc-gen-validate@d6164de4910977d3c3c8dbd9299b5064ea9e7c2b (module  )
exec 'go mod init -modfile=/Users/nacx/src/go/src/github.com/tetrateio/ngac/.bingo/protoc-gen-validate-e.tmp.mod _'
exec 'go get -modfile=/Users/nacx/src/go/src/github.com/tetrateio/ngac/.bingo/protoc-gen-validate-e.tmp.mod -d github.com/envoyproxy/protoc-gen-validate@d6164de4910977d3c3c8dbd9299b5064ea9e7c2b'
exec 'go env GOPATH'
exec 'go mod init -modfile=/Users/nacx/src/go/src/github.com/tetrateio/ngac/.bingo/protoc-gen-validate.tmp.mod _'
exec 'go mod download -x -modfile=/Users/nacx/src/go/src/github.com/tetrateio/ngac/.bingo/protoc-gen-validate.tmp.mod'
exec 'go list -modfile=/Users/nacx/src/go/src/github.com/tetrateio/ngac/.bingo/protoc-gen-validate.tmp.mod -f={{.Name}} github.com/envoyproxy/protoc-gen-validate'
Error: install: build versioned: ../../../../../pkg/mod/github.com/envoyproxy/protoc-gen-validate@v0.0.15-0.20190405222122-d6164de49109/gogoproto/gogo.pb.go:15:8: no required module provides package github.com/golang/protobuf/proto; to add it:
	go get github.com/golang/protobuf/proto
../../../../../pkg/mod/github.com/envoyproxy/protoc-gen-validate@v0.0.15-0.20190405222122-d6164de49109/gogoproto/gogo.pb.go:18:8: no required module provides package github.com/golang/protobuf/protoc-gen-go/descriptor; to add it:
	go get github.com/golang/protobuf/protoc-gen-go/descriptor
../../../../../pkg/mod/github.com/envoyproxy/protoc-gen-validate@v0.0.15-0.20190405222122-d6164de49109/templates/cc/register.go:10:2: no required module provides package github.com/golang/protobuf/ptypes; to add it:
	go get github.com/golang/protobuf/ptypes
../../../../../pkg/mod/github.com/envoyproxy/protoc-gen-validate@v0.0.15-0.20190405222122-d6164de49109/validate/validate.pb.go:10:8: no required module provides package github.com/golang/protobuf/ptypes/duration; to add it:
	go get github.com/golang/protobuf/ptypes/duration
../../../../../pkg/mod/github.com/envoyproxy/protoc-gen-validate@v0.0.15-0.20190405222122-d6164de49109/validate/validate.pb.go:11:8: no required module provides package github.com/golang/protobuf/ptypes/timestamp; to add it:
	go get github.com/golang/protobuf/ptypes/timestamp
../../../../../pkg/mod/github.com/envoyproxy/protoc-gen-validate@v0.0.15-0.20190405222122-d6164de49109/templates/java/register.go:12:2: no required module provides package github.com/iancoleman/strcase; to add it:
	go get github.com/iancoleman/strcase
../../../../../pkg/mod/github.com/envoyproxy/protoc-gen-validate@v0.0.15-0.20190405222122-d6164de49109/templates/shared/context.go:11:2: no required module provides package github.com/lyft/protoc-gen-star; to add it:
	go get github.com/lyft/protoc-gen-star
../../../../../pkg/mod/github.com/envoyproxy/protoc-gen-validate@v0.0.15-0.20190405222122-d6164de49109/templates/cc/register.go:14:2: no required module provides package github.com/lyft/protoc-gen-star/lang/go; to add it:
	go get github.com/lyft/protoc-gen-star/lang/go
: error while running command 'go build -modfile=/Users/nacx/src/go/src/github.com/tetrateio/ngac/.bingo/protoc-gen-validate.tmp.mod -o=/Users/nacx/src/go/bin/protoc-gen-validate-v0.0.15-0.20190405222122-d6164de49109 github.com/envoyproxy/protoc-gen-validate'; err: exit status 1
protoc-gen-validate.mod: getting github.com/envoyproxy/protoc-gen-validate@d6164de4910977d3c3c8dbd9299b5064ea9e7c2b
main.get
	/Users/nacx/src/go/src/github.com/bwplotka/bingo/get.go:296
main.main.func2
	/Users/nacx/src/go/src/github.com/bwplotka/bingo/main.go:156
main.main.func5
	/Users/nacx/src/go/src/github.com/bwplotka/bingo/main.go:247
github.com/oklog/run.(*Group).Run.func1
	/Users/nacx/src/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38
runtime.goexit
	/usr/local/opt/go/libexec/src/runtime/asm_amd64.s:1371
get
main.main.func2
	/Users/nacx/src/go/src/github.com/bwplotka/bingo/main.go:157
main.main.func5
	/Users/nacx/src/go/src/github.com/bwplotka/bingo/main.go:247
github.com/oklog/run.(*Group).Run.func1
	/Users/nacx/src/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38
runtime.goexit
	/usr/local/opt/go/libexec/src/runtime/asm_amd64.s:1371
get command failed
main.main
	/Users/nacx/src/go/src/github.com/bwplotka/bingo/main.go:255
runtime.main
	/usr/local/opt/go/libexec/src/runtime/proc.go:225
runtime.goexit
	/usr/local/opt/go/libexec/src/runtime/asm_amd64.s:1371
```
We should add that one as an e2e test case as well.

Reply:
Is that failing after my PR too? (:

Reply:
Your commit fixes this :)